src/components/AmountFilter.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import { Button, Popover, InputLabel, Grid } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import RemoveIcon from '@material-ui/icons/Remove';
+import AmountTextField from './AmountTextField';
+
+const useStyles = makeStyles((theme) => ({
+  buttonStyle: {
+    margin: theme.spacing(1),
+    maxWidth: 120,
+    height: 57,
+  },
+  labelStyle: { paddingLeft: theme.spacing(1) },
+  dashStyle: { margin: '45px auto auto' },
+  filterStyle: { margin: '10px 0' },
+}));
+
+export default function AmountFilter() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [values, setValues] = useState({});
+
+  const handleClick = (event) => setAnchorEl(event.currentTarget);
+
+  const handleChange = (index, val) =>
+    setValues({ ...values, [index]: parseInt(val, 10) || '' });
+
+  function displayFilter(label, value, onChange) {
+    return (
+      <Grid item className={classes.filterStyle}>
+        <InputLabel className={classes.labelStyle}>{label}</InputLabel>
+        <AmountTextField
+          disabled={false}
+          variation="outlined"
+          {...{ value, onChange }}
+        />
+      </Grid>
+    );
+  }
+
+  return (
+    <>
+      <Button
+        variant="outlined"
+        className={classes.buttonStyle}
+        onClick={handleClick}
+        endIcon={<ArrowDropDownIcon />}>
+        Amount
+      </Button>
+      <Popover
+        id="amount-filter"
+        open={Boolean(anchorEl)}
+        anchorEl={anchorEl}
+        onClose={() => setAnchorEl(null)}
+        anchorOrigin={{ vertical: 'top', horizontal: 'left' }}
+        transformOrigin={{ vertical: -5, horizontal: 'left' }}>
+        <Grid container>
+          {displayFilter('Min Amount', values.min, (e) =>
+            handleChange('min', e.target.value)
+          )}
+          <Grid item className={classes.dashStyle}>
+            <RemoveIcon />
+          </Grid>
+          {displayFilter('Max Amount', values.max, (e) =>
+            handleChange('max', e.target.value)
+          )}
```

Comment:
I thought for these you would reuse the [`AmountTextField`](https://github.com/beyondhb1079/s4us/blob/main/src/components/AmountTextField.jsx) component. If it's not good enough yet then I'd adjust that instead so it can be reused. Otherwise if we change how we display/manage amounts in one component we'll have to remember to change it in the `displayFilter` component here too.

Reply:
I did in the `displayFilter` function above. I just extracted it into another function to make it easier to read, since it takes about 8 lines of code to display the `AmountTextField` with its label.

Reply:
Ah I see. Hmm, I think this may be a case of over-optimization. Helper function components are nice when you can logically separate JSX elements that make up component. In this case though, it's a parent `<Grid item>` element with two child elements. The `<Grid item>` element should probably be wherever its `<Grid item>` siblings are or refactored such that some mapping is done if the children are similar.